public/app/features/storage/UploadView.tsx

```diff
@@ -84,8 +88,25 @@ export const UploadView = ({ folder, onUpload }: Props) => {
         {error.message !== '' ? <p>{error.message}</p> : Boolean(file) ? <Preview /> : <FileDropzoneCustomChildren />}
       </FileDropzone>
 
+      {file && filenameExists && (
+        <div className={styles.alert}>
+          <Alert title={`${file.name} already exists`} severity="error">
+            <Checkbox
+              value={overwriteExistingFile}
+              onChange={() => setOverwriteExistingFile(!overwriteExistingFile)}
+              label="Overwrite existing file"
+            />
+          </Alert>
+        </div>
+      )}
+
       <ButtonGroup>
-        <Button className={styles.button} variant={'primary'} disabled={!file} onClick={doUpload}>
+        <Button
+          className={styles.button}
+          variant={'primary'}
+          disabled={!file || (filenameExists && !overwriteExistingFile)}
```

Comment:
Nit but may be a bit cleaner to separate this out into a separate bool above the return, maybe something like `isUploadDisabled`